modules/auth_email/utils/link_assertions.ts

```diff
@@ -0,0 +1,107 @@
+import { RuntimeError, ScriptContext } from "../module.gen.ts";
+import {
+	IDENTITY_INFO_LINK,
+	IDENTITY_INFO_PASSWORD,
+	IDENTITY_INFO_PASSWORDLESS,
+} from "./provider.ts";
+
+export async function ensureNotAssociated(
```

Comment:
repost: this is very very expensive, it does at least 3 round trips to the database before doing anything else in the request. we should expose a batch api on identities module for this.

to answer the q: it should be very easy to pass multiple identity types to the query so it can be configurable. but also, that won't be a breaking change, so go for the simpler option.